alibi_detect/od/backend/__init__.py

```diff
@@ -0,0 +1,9 @@
+from alibi_detect.utils.missing_optional_dependency import import_optional
+
+KNNTorch = import_optional('alibi_detect.od.backend.torch.knn', ['KNNTorch'])
+PValNormalizerTorch, ShiftAndScaleNormalizerTorch, TopKAggregatorTorch, AverageAggregatorTorch, \
```

Comment:
Re the `AverageAggregator` wrapper, yep that is a much neater (🙂) version of what I had in mind. Just so the user doesn't have to do `from alibi_detect.od.backend import ...`, which is confusing since for drift detectors they instead do `from alibi_detect.cd.pytorch import ...`. The alternative of just exposing `AverageAggregatorTorch` in `od.__init__` for now also seems reasonable, and we can add a wrapper if/when more backends are added in the future. However, with this option, I still wonder if its worth exposing `AverageAggregatorTorch` as `AverageAggregator` in `od.__init__`? Or is there a reason to require the user to explicitly select the `...Torch` classes?

Reply:
@mauicv looks like this comment is resolved by your changes, but can I just repeat what I think the plan so we can check we're on the same page?

1. `KNN` is currently private i.e. `alibi_detect.od._knn import KNN`
2. Can import it via `from alibi_detect.experimental.od import KNN`
3. This will be promoted to `alibi_Detect.od import KNN` when we are ready for release.
4. We will only think about making the `from alibi.od.pytorch` public, or adding a wrapper, if we add more backends in the future?